packages/app/features/top/PostScreen.tsx

```diff
@@ -64,7 +64,7 @@ export default function PostScreen(props: Props) {
   }, [post, threadPosts]);
 
   const markRead = useCallback(() => {
-    if (channel && post && threadPosts) {
+    if (channel && post && threadPosts && threadPosts.length > 0) {
```

Comment:
There we have it!

Reply:
indeed! Though we also found repeated valid channel reads occasionally error for unknown reasons